az-hdfs-viewer/src/main/java/azkaban/viewer/hdfs/ParquetFileViewer.java

```diff
@@ -58,15 +52,14 @@ public String getName() {
   }
 
   @Override
-  public Set<Capability> getCapabilities(FileSystem fs, Path path)
-      throws AccessControlException {
+  public Set<Capability> getCapabilities(FileSystem fs, Path path) {
     if (logger.isDebugEnabled()) {
       logger.debug("Parquet file path: " + path.toUri().getPath());
     }
 
     AvroParquetReader<GenericRecord> parquetReader = null;
     try {
-      parquetReader = new AvroParquetReader<GenericRecord>(path);
+      parquetReader = new AvroParquetReader<>(path);
```

Comment:
I found a [commit](https://github.com/ypadron-in/azkaban/commit/d8532ea229fbe8132e485dbf5832d15d97f74ff3) (not yet finished apparently) in my local repo that was supposed to fix below issue:
https://github.com/azkaban/azkaban/pull/2861
> 
The issue is: Files that should be handled by the TextFileViewer (like JSON files) are now incorrectly handled by the ParquetFileViewer.

Could you please take a look at the commit? There are a couple of tests that you can leverage perhaps and more importantly there is a comment that suggests that we need to read the first record of the file to actually prevent the issue.